src/pointer.rs

```diff
@@ -595,9 +641,13 @@ impl PointerBuf {
     }
 
     /// Merges two `Pointer`s by appending `other` onto `self`.
-    pub fn append(&mut self, other: &PointerBuf) -> &PointerBuf {
+    pub fn append<P: AsRef<Pointer>>(&mut self, other: P) -> &PointerBuf {
+        self._append(other.as_ref())
+    }
+
+    fn _append(&mut self, other: &Pointer) -> &PointerBuf {
         if self.is_root() {
-            self.0.clone_from(&other.0);
+            self.0 = other.0.to_string();
         } else if !other.is_root() {
             self.0.push_str(&other.0);
         }
```

Comment:
Why the split?

Reply:
Oh, because I made `append` generic and I was toying around trying to understand `#[inline(never)]` a bit better. I just forgot to switch it back.

Reply:
Thanks for catching it. I'll move it back.